h2/src/main/org/h2/result/ResultTempTable.java

```diff
@@ -304,12 +309,7 @@ private Row convertToRow(Value[] values) {
     }
 
     private Cursor find(Row row) {
-        if (index == null) {
-            // for the case "in(select ...)", the query might
-            // use an optimization and not create the index
-            // up front
-            createIndex();
-        }
```

Comment:
maybe move this comment to the getIndex() method

Reply:
OK, I'll place some comments later.